pages/index.js

```diff
@@ -1,14 +1,27 @@
 import Head from "next/head";
 import styles from "styles/Home.module.css";
 import Layout from "components/Layout"
+import HomeProducts from "components/HomeProducts"
 
-export default function Home() {
+export default function Home(props) {
   return (
     <Layout>
       <Head>
         <title>Ecommerce Coffee</title>
       </Head>
       <div className={styles.container}></div>
+      <HomeProducts coffees={props.firstNine} itemsLeft={props.itemsLeft} />
```

Comment:
items left can be calculated, even if we don't use the backend to retrieve the data, you can use a limit variable to control how many products you show first, and  if you increment the limit to show and this equal or greather than the products array length you need that there is not more products to show

Reply:
You mean implementing a kind of _Limit-Ofset_?

Reply:
yes, let's take the following example, that take as many numbers from the array until reach the limit 

```js

function getNumbers(limit) {
   const items = [ 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15 ];
 
  return items.slice(0, limit);
}

let limit = 3;

getNumbers(limit); // take until limit = three numbers [1, 2, 3]

limit+=3;


getNumbers(limit); // take until limit = six numbers  [1, 2, 3, 4, 5, 6]

```

Reply:
@Jsifontez If you don't want to take repeated numbers you can work with a start off-set

```
function getNumbers(start, limit) {
   const items = [ 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15 ];
 
  return items.slice(start, limit);
}

let offset = 0;
let limit = 3;

getNumbers(offset, limit + offset); // [ 1,2,3]

offset+= 3;

getNumbers(offset, limit + offset) // [4,5, 6]

// And so on
```

Reply:
@Jsifontez You know that there is not more numbers to show if limit >= numbers.length